index.bs

```diff
@@ -2166,8 +2237,125 @@ Methods</h4>
 		<div>
 			<em>Return type:</em> {{Promise}}&lt;{{undefined}}&gt;
 		</div>
+
+		: <dfn>setSinkId((DOMString or AudioSinkOptions) sinkId)</dfn>
+		::
+			Sets the identifier of an output device. When this method is invoked, the
+			user agent MUST run the following steps:
+
+			<div algorithm="AudioContext.setSinkId()">
+				1. Let |sinkId| be the method's first argument.
+
+				1. Let |validationResult| be the return value of
+					<a href="#validating-sink-identifier">sink identifier validation</a>
+					of |sinkId|.
+				
+				1. If |validationResult| is not true, return a promise rejected with
+					|validationResult|. Abort these steps.
+
+				1. Let |p| be a new promise.
+
+				1. Send a <a>control message</a> with |p| and |sinkId| to start
+					processing.
+
+				1. Return |p|.
+			</div>
+
+			<div algorithm="run a control message for setSinkId">
+				Sending a <a>control message</a> to start processing during
+				{{AudioContext/setSinkId()}} means executing the following steps:
+
+				1. Let |p| be the promise passed into this algorithm.
+
+				1. Let |sinkId| be the sink identifier passed into this algorithm.
+
+				1. Let |wasRunning| be true.
+
+				1. Set |wasRunning| to false if the {{[[rendering thread state]]}} on
+					the {{AudioContext}} is <code>"suspended"</code>.
+
+				1. Pause the renderer after processing the current render quantum.
+
+				1. Attempt to <a>release system resources</a>.
+
+				1. If |wasRunning| is true:
+
+					1. Set the {{[[rendering thread state]]}} on the {{AudioContext}} to
```

Comment:
It feels like this (and it's "resume" counterpart below) should be spun up in their own algorithm, but we can do it later.

Reply:
ACK